mentat/code_context.py

```diff
@@ -186,15 +195,15 @@ def refresh(self, replace_paths: bool | None = False):
             if replace_paths and not self.settings.paths:
                 self.settings.paths = self.diff_context.files
         except UserError as e:
-            cprint(str(e), "light_yellow")
-            exit()
+            await stream.send(str(e), color="light_yellow")
+            exit()  # NOTE: should this be an exit()?
```

Comment:
I was copying the pattern from `expand_paths` - which I see still exists but is now in `mentat/terminal/client.py`. Anyway it's not functioning correctly - if you give it an invalid diff, right now it just closes. 

I replaced the call to `exit()` this with a MentatError, and move the initialization of `CodeContext` to `Session._main`, right before Conversation. Now it gives you an informative message and then closes.